domain/src/main/kotlin/org/bmsk/domain/model/StringAdditionCalculator.kt

```diff
@@ -0,0 +1,45 @@
+package org.bmsk.domain.model
+
+class StringAdditionCalculator(private val stringFormula: String) {
+
+    private val realStringFormula: String = extractRealStringFormula()
+    private val customSeparators: Separators = extractCustomSeparators()
+    private val allSeparators: Separators = defaults + customSeparators
+
+    init {
+        StringAdditionValidator().validateFormat(realStringFormula, allSeparators)
+    }
+
+    fun calculate(): Int {
+        return separate().sum()
+    }
+
+    internal fun separate(): List<Int> {
+        realStringFormula.ifBlank { return emptyList() }
+        val pattern = allSeparators.toRegex()
+        return realStringFormula.split(pattern).map { it.trim().toInt() }
+    }
+
+    private fun extractCustomSeparators(): Separators {
+        val result = Regex("//(.)\n(.*)").find(stringFormula)
+        return if (result != null) {
+            Separators(listOf(Separator(result.groupValues[1])))
+        } else {
+            Separators(emptyList())
+        }
+    }
+
+    private fun extractRealStringFormula(): String {
+        return if (stringFormula.contains("\n")) {
+            stringFormula.split("\n")[1]
+        } else {
+            stringFormula
+        }
+    }
```

Comment:
아래처럼 바로 =을 통해 리턴해줄수도 있습니다.

```suggestion
    private fun extractRealStringFormula(): String = if (stringFormula.contains("\n")) {
            stringFormula.split("\n")[1]
        } else {
            stringFormula
        }
```

Reply:
리뷰어님은 =을 통해 바로 리턴하는 것을 선호하실까요?
저는 `return`을 적어주는 것을 가독성이 더 좋은 것 같아서 선호했습니다.

물론 =으로 바로 표현하는 것이 간결하다는 장점이 있는 것 같습니다.

Reply:
타입 없이 =으로 표현하면 다소 가독성이 좋지 않은 측면이 있지만 타입이 명시되어 있다면 =으로 간결하게 표현하는 것을 선호하는 편입니다. 이 부분은 중요한 것은 아니며 코틀린의 문법이라 말씀드렸습니다.

취향에 따라 적용하시면 될 것 같습니다.